File: src/main/java/guru/xgm/angelscript/formatting/panel/ZincCodeStyleMainPanel.java
package guru.xgm.angelscript.formatting.panel;

import com.intellij.application.options.TabbedLanguageCodeStylePanel;
import com.intellij.psi.codeStyle.CodeStyleSettings;
import com.intellij.psi.codeStyle.CodeStyleSettingsProvider;
import guru.xgm.angelscript.lang.ZincLanguage;

public class ZincCodeStyleMainPanel extends TabbedLanguageCodeStylePanel {
    public ZincCodeStyleMainPanel(CodeStyleSettings currentSettings, CodeStyleSettings settings) {
        super(ZincLanguage.INSTANCE, currentSettings, settings);
    }

    @Override
    protected void initTabs(CodeStyleSettings settings) {
        super.initTabs(settings);
        for (CodeStyleSettingsProvider provider : CodeStyleSettingsProvider.EXTENSION_POINT_NAME.getExtensionList()) {
            if (provider.getLanguage() == ZincLanguage.INSTANCE && !provider.hasSettingsPage()) {
                createTab(provider);
            }
        }
    }
}

File: src/main/kotlin/guru/xgm/language/zinc/formatting/panel/ZincCodeStyleMainPanel.kt
package guru.xgm.language.zinc.formatting.panel

import com.intellij.application.options.TabbedLanguageCodeStylePanel
import com.intellij.psi.codeStyle.CodeStyleSettings
import com.intellij.psi.codeStyle.CodeStyleSettingsProvider
import guru.xgm.language.zinc.lang.ZincLanguage

class ZincCodeStyleMainPanel(currentSettings: CodeStyleSettings?, settings: CodeStyleSettings?) :
    TabbedLanguageCodeStylePanel(ZincLanguage.instance, currentSettings, settings!!) {
    override fun initTabs(settings: CodeStyleSettings) {
        super.initTabs(settings)
        for (provider in CodeStyleSettingsProvider.EXTENSION_POINT_NAME.extensionList) {
            if (provider.language === ZincLanguage.instance && !provider.hasSettingsPage()) {
                createTab(provider)
            }
        }
    }
}
